pkg/instance/instance.go

```diff
@@ -9,8 +12,11 @@ import (
 
 // Info represents a instance that is meant to be joined to the cluster
 type Info struct {
-	// Address is the network address of the instance
+	// Address is the network address of the instance as specified by the associated ConfigMap entry.
+	// Must be an IPv4 address or a DNS name that resolves to one.
 	Address string
+	// IPv4Address is the IPv4 address associated with the instance's given Address. May be the same value.
+	IPv4Address string
```

Comment:
Do we need separate entries? Do we really need the DNS entry stored? IOW can we just use the IPv4 address everywhere? PS: this is a question as I have not looked at all the places where `Address` and whether it needs to be storied. If you find using the IPv4 address everywhere is involved, then we can leave it for a follow up PR. Otherwise just store the IPv4 address in `Address`.

Reply:
The two reservations I had with consolidating to 1 entry were
1. Error messages and logs would be hard to follow. For example, a DNS entry in the ConfigMap would be resolved to some IP, and only this IP would be present in all output messages which could be tricky to understand.
2. I am not sure how using the resolved IP address would affect [the CSR logic ](https://github.com/openshift/windows-machine-config-operator/blob/6901ac3ed2891b7627a1431ab66703358b902671/pkg/csr/csr.go#L320-L336)

Would suggest a follow-up PR.

Reply:
Story created here https://issues.redhat.com/browse/WINC-675.